packages/components/Input/src/Input.types.ts

```diff
@@ -0,0 +1,247 @@
+
+export const input = 'Input';
+
+export interface InputCoreTokens extends LayoutTokens, IBorderTokens, IColorTokens, FontTokens {
+  /**
+   * The icon color.
+   */
+  iconColor?: ColorValue;
+
+  /**
+   * The size of the icon.
+   */
+  iconSize?: number;
+
+  /**
+   * The dismiss icon color.
+   */
+  accessoryIconColor?: ColorValue;
+
+  /**
+   * The size of the dismiss icon.
+   */
+  accessoryIconSize?: number;
+
+  /**
+   * The color of the assistive text.
+   */
+  assistiveTextColor?: ColorValue;
+
+  /**
+   * The font style of the assistive text.
+   */
+  assistiveTextFont?: FontTokens;
+
+  /**
+   * The font style of the input box.
+   */
+  inputTextFont?: FontTokens;
+
+  /**
+   * The color of the input text.
+   */
+  inputTextColor?: ColorValue;
+
+  /**
+   * The font style of the secondary text.
+   */
+  secondaryTextFont?: FontTokens;
+
+  /**
+   * The color of the secondary text.
+   */
+  secondaryTextColor?: ColorValue;
+
+  /**
+   * The width of the divider.
+   */
+  strokeWidth?: number;
+
+  /**
+   * The color of the divider.
+   */
+  strokeColor?: ColorValue;
+
+  /**
+   * The color of the cursor.
+   */
+  cursorColor?: ColorValue;
+
+  /**
+   * The space between the icon and the rest of the content.
+   */
+  spacingIconContent?: number;
+
+  /**
+   * The space between the input and the secondary text.
+   */
+  spacingInputSecondary?: number;
+
+  /**
+   * The space between the dismiss icon and the left content.
+   */
+  spacingAccessoryIconStart?: number;
+
+  /**
+   * The input's vertical margin.
+   */
+  spacingInputVertical?: number;
+
+  /**
+   * The assistive text's vertical margin.
+   */
+  spacingAssistiveTextVertical?: number;
+
+  /**
+   * The assistive text's start margin. Used to align it with input box.
+   */
+  spacingAssistiveTextStart?: number;
+
+  /**
+   * The label's start margin. Used to align it with input box.
+   */
+  spacingLabelStart?: number;
+
+  /**
+   * The label's top margin.
+   */
+  spacingLabelTop?: number;
+}
+
+export interface InputTokens extends InputCoreTokens {
+  /**
+   * States that can be applied to an input.
+   */
+  hasIcon?: InputTokens;
+  error?: InputTokens;
+  focused?: InputTokens;
+  typing?: InputTokens;
+  filled?: InputTokens;
+}
+
+export interface InputProps extends PressableFocusProps {
+  /*
+   * Source URL or name of the icon to show on the input.
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * The width of the button.
+   */
+  width?: ViewStyle['width'];
+
+  /*
+   * Source URL or name of the accessory icon to show on the input.
+   * Shown while input has text in it. Not shown while input is used as a control component.
+   *
+   * @default - dismiss icon is shown.
+   * Set as null to disabled this icon.
+   */
+  accessoryIcon?: IconSourcesType | null;
+
+  /**
+   * A callback to call on accessoryIcon click event.
```

Comment:
,maybe "Callback that is called when the text input's accessoryIcon is clicked" - refencing the RN handler docs

Reply:
Have followed similar language to that used already in our repo (like onClick for button)